src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -763,32 +807,32 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebooks(int accountId, cons
             QString currDeviceCalendarId = notebook->pluginName().mid(7);
             if (m_serverCalendarIdToCalendarInfo[accountId].contains(currDeviceCalendarId)) {
                 // the server-side calendar exists on the device.
-                // we don't need to purge it, but we may need to update its summary/color details.
-                deviceCalendarIds.append(currDeviceCalendarId);
-                if (notebook->name() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary
-                        || notebook->color() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color
-                        || notebook->description() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description
-                        || notebook->isReadOnly()) {
-                    // calendar information changed server-side.
-                    notebook->setIsReadOnly(false);
-                    notebook->setName(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary);
-                    notebook->setColor(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color);
-                    notebook->setDescription(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description);
-                    m_storage->updateNotebook(notebook);
-                    m_storageNeedsSave = true;
+                if (needCleanSync) {
+                    // we are performing a clean sync cycle.
+                    // we will eventually delete and then insert this notebook.
+                    SOCIALD_LOG_DEBUG("queuing clean sync of local calendar" << notebook->name() << currDeviceCalendarId << "for Google account:" << accountId);
+                    m_serverCalendarIdToCalendarInfo[accountId][currDeviceCalendarId].change = GoogleCalendarSyncAdaptor::CleanSync;
```

Comment:
needCleanSync == !wasLastSyncSuccesfull(). So here local modifications will be gone after two network errors?

As another detail, I'm not sure I fully follow the need to postpone notebook modifications. Here full server side notebook list should be known, so are there bad side-effects if notebook changes are done even if processing events is later aborted?